responses/matchers.py

```diff
@@ -402,14 +403,31 @@ def header_matcher(
     :return: (func) matcher
     """
 
+    def _compare_with_regex(request_headers: Union[Dict[Any, Any], Any]) -> bool:
+        if strict_match and len(request_headers) != len(headers):
+            return False
+
+        for k, v in headers.items():
+            if request_headers.get(k) is not None:
```

Comment:
what does RFC say about headers. Can a header officially be equal None ?

Reply:
```
Field values containing CR, LF, or NUL characters are invalid and dangerous ...
```

-- from RFC 9110 [here](https://datatracker.ietf.org/doc/html/rfc9110#name-field-values)

I'd say its safe to assume that any header will not be None

Reply:
yes, confirm

Headers do not support None and should be at least a whitespace